libraries/Microsoft.Bot.Builder.Core.State/IStateStorageProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Core.State
+{
+    public interface IStateStorageProvider
+    {
+        IStateStorageEntry CreateNewEntry(string stateNamespace, string key);
+
+        /// <summary>
+        /// Loads all state entries under the specified <paramref name="stateNamespace">partition</paramref>.
+        /// </summary>
+        /// <param name="stateNamespace"></param>
+        /// <returns></returns>
+        Task<IEnumerable<IStateStorageEntry>> Load(string stateNamespace);
+
+        /// <summary>
+        /// Loads a single piece of state, identified by its <paramref name="key"/> from the given <paramref name="stateNamespace">partition</paramref>.
+        /// </summary>
+        /// <param name="stateNamespace"></param>
+        /// <param name="key"></param>
+        /// <returns></returns>
+        Task<IStateStorageEntry> Load(string stateNamespace, string key);
```

Comment:
What's the expected behavior if a object is missing?

Reply:
You get a `null` back in C# indicating that no storage entry exists for the specified namespace/key. So I guess for JS you can decide if `null` or `undefined` makes more sense.